enclave/server.ts

```diff
@@ -57,13 +57,59 @@ const nStates = new ethers.Contract(
 let b: Board;
 
 /*
- * Adjust internal state based on claimed move & notifies all users to
- * update their local views.
+ * Propose move to enclave. In order for the move to be solidified, the enclave
+ * must respond to a leaf event.
  */
-function move(tFrom: any, tTo: any, uFrom: any, uTo: any) {
-    b.setTile(Tile.fromJSON(uFrom));
-    b.setTile(Tile.fromJSON(uTo));
-    io.sockets.emit("updateDisplay");
+async function propose(socket: Socket, uFrom: any, uTo: any) {
+    const uFromAsTile = Tile.fromJSON(uFrom);
+    const hUFrom = uFromAsTile.hash();
+    const uToAsTile = Tile.fromJSON(uTo);
+    const hUTo = uToAsTile.hash();
+
+    const digest = utils.solidityKeccak256(
+        ["uint256", "uint256"],
+        [hUFrom, hUTo]
+    );
+    const sig = await signer.signMessage(utils.arrayify(digest));
+
+    socket.emit("proposeResponse", sig, uFrom, uTo);
+}
+
+/*
+ * Alert enclave that solidity accepted new states into the global state.
+ * Enclave should confirm by checking NewLeaf events and change it's own belief.
+ *
+ * [TODO]: automate this with an Alchemy node.
```

Comment:
same question as above with sending "ping" on the client side 

it does seem like your plan to take the client out of this loop completely, just want to make sure

Reply:
yes